project-belongs-to-user

From: Bryan Larsen <bryan@larsen.st>

# Project ownership

The next goal for Agility is to move to a full multi-user application, where users can create their own projects and control who has access to them. Rather than make this change in one go, we'll start with a small change that doesn't do much by itself, but is a step in the right direction: making projects be owned by users.

Add the following to the Project model:

SHOW_PATCH

There's a Hobo extension there: `:creator => true` tells Hobo that when creating one of these things, the `owner` association should be automatically set up to be the user doing the create.
---
 app/models/project.rb |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/app/models/project.rb b/app/models/project.rb
index 09933f0..cac656a 100644
--- a/app/models/project.rb
+++ b/app/models/project.rb
@@ -13,6 +13,8 @@ class Project < ActiveRecord::Base
 
   children :stories
 
+  belongs_to :owner, :class_name => "User", :creator => true, :inverse_of => :project
+
   # --- Permissions --- #
 
   def create_permitted?
